src/core/src/energy/global/ewald.rs

```diff
@@ -759,15 +759,13 @@ impl GlobalPotential for SharedEwald {
         return real + self_e + kspace + molecular;
     }
 
-    fn forces(&self, system: &System) -> Vec<Vector3D> {
+    fn forces(&self, system: &System, forces: &mut [Vector3D])  {
```

Comment:
Be cause we no longer allocate the data ourself, we will need to assert that the slice has the right length whenever we get one as a parameter.